source/Optimizers/nelder_mead.d

```diff
@@ -1,21 +1,18 @@
 module source.optimizers.nelder_mead;
 
-import std.algorithm: minElement, topN, minIndex, topNIndex;
+import std.algorithm;
 import std.conv: to;
 import std.functional: toDelegate;
 import std.math: abs, pow, sqrt;
 import std.range: enumerate;
+import std.stdio: writeln, write;
+import std.typecons: Yes;
 
 import source.Matrix;
 import source.NeuralNetwork;
 import source.Parameter;
 
 
-version(unittest) {
-    import std.stdio: writeln, write;

```

Comment:
I agree this should just be included up top for very common dependencies -- std.stdio write is already loaded and being linked to by other files in the project so it won't bloat binary size or compile time by just including them